extensions/libraries/redcore/api/payment/helper.php

```diff
@@ -759,6 +763,27 @@ public static function updatePaymentStatus($paymentId)
 		return true;
 	}
 
+	/**
+	 * Check if ePay is using  instant capture
+	 *
+	 * @param   object	$payment	Payment object
+	 *
+	 * @return  boolean
+	 */
+	public static function isInstantCapture($payment)
+	{
+		$plugin 		= RApiPaymentHelper::getPaymentParams($payment->payment_name, $payment->extension_name, $payment->owner_name);
+		$instantcapture = false;
+		$onlinePayments = array('epay', 'paypal', 'quickpay');
+
+		if ($plugin && in_array($plugin->element, $onlinePayments))
```

Comment:
I am not sure that in_array($plugin->element, $onlinePayments) is needed if you have fixed $plugin->params->get('instantcapture', 0) parameter that we will use in all our plugins.

Reply:
@Kixo But Paypal and Quickpay also can do partial capture, but those 2 plugins don't have the `instantcapture` parameter.

Reply:
Actually I didn't understand your "...if you have fixed $plugin->params->get('instantcapture', 0) parameter that we will use in all our plugins."
Did you mean that the `if (in_array($plugin->element, $onlinePayments))` check around `$instantcapture = (boolean) $plugin->params->get('instantcapture', 0);` is unnecesary and that `$instantcapture = (boolean) $plugin->params->get('instantcapture', 0);` by itself is fine?

Reply:
I tested with bank transfer. In order, I still have the capture payment option :(

Reply:
![image](https://user-images.githubusercontent.com/20809372/46528790-d3f21800-c8be-11e8-9fc4-b3fe6f415232.png)
Paypal can handle capture payments and partial captures, yet I get this error :(

I think I just discovered a new load of problems